projects/client/RabbitMQ.Client/src/client/impl/ConsumerWorkService.cs

```diff
@@ -49,46 +40,34 @@ public void StopWork()
         class WorkPool
         {
             readonly ConcurrentQueue<Action> actions;
-            readonly AutoResetEvent messageArrived;
-            readonly TimeSpan waitTime;
```

Comment:
Much better to solve with SemaphoreSlim rather than an AutoResetEvent. Cleaner code and more reliable.